libs/lib-linker/src/lib/lib-linker.ts

```diff
@@ -0,0 +1,52 @@
+export type LinkerResult =
+  | 'OK'
+  | 'IDENTITY_UNKNOWN'
+  | 'IDENTITY_REGISTERED_ALREADY'
+  | 'IDENTITY_LINKED'
+  | 'IDENTITY_IDENTICAL'
+  | 'TOKEN_EXPIRED'
+  | 'UNKNOWN_ERROR'
+  | 'MESSAGE_SENT_TO_SUPPORT';
+
+export const LinkerResult = {
+  OK: 'OK' as LinkerResult,
+  IDENTITY_UNKNOWN: 'IDENTITY_UNKNOWN' as LinkerResult,
+  IDENTITY_REGISTERED_ALREADY: 'IDENTITY_REGISTERED_ALREADY' as LinkerResult,
+  IDENTITY_LINKED: 'IDENTITY_LINKED' as LinkerResult,
+  IDENTITY_IDENTICAL: 'IDENTITY_IDENTICAL' as LinkerResult,
+  TOKEN_EXPIRED: 'TOKEN_EXPIRED' as LinkerResult,
+  MESSAGE_SENT_TO_SUPPORT: 'MESSAGE_SENT_TO_SUPPORT' as LinkerResult,
+  UNKNOWN_ERROR: 'UNKNOWN_ERROR' as LinkerResult,
+};
+
+export function openLinkerWindow(
+  linkerUrl: string,
+  formerToken: string,
+  idpFilter?: string[]
+): Window | null {
+  let url = linkerUrl + '?formerToken=' + formerToken;
+  if (idpFilter) {
+    if (idpFilter.length !== 0) {
+      url += '&idphint=' + idpFilter.toString();
+    }
+  }
+  const width = 600;
+  const height = 600;
+
+  const top = (window.top as Window).outerHeight / 2 + (window.top as Window).screenY - height / 2;
```

Comment:
Create a variable for the `window.top` so there is no need to cast multiple times.